From 2648a4064aff7ae82e74b8de6ade93dc9c04ce3b Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 12 May 2017 18:22:02 +0500
Subject: [PATCH] autoopts/mk-tpl-config.sh: fix shebang length

The shebang length cannot be more than 128 charaters
and the change
http://cgit.openembedded.org/openembedded-core/commit/meta/classes/base.bbclass?id=fa764a403da34bb0ca9fa3767a9e9dba8d685965
requires that a user needs to make sure the HOSTTOOLS
directory is under 128 characters in order to get
the build through otherwise this causes
bad interpreter: Permission denied
when the builddir path is too long.
This change now follows the path to actual host
executable to be put in the final scripts that
are generated so the shebang remains correct in
most situations.

Upstream-Status: Inappropriate

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 autoopts/mk-tpl-config.sh | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/autoopts/mk-tpl-config.sh b/autoopts/mk-tpl-config.sh
index c4708a2..7de546b 100755
--- a/autoopts/mk-tpl-config.sh
+++ b/autoopts/mk-tpl-config.sh
@@ -98,11 +98,14 @@ fix_scripts() {
         st=`sed 1q $f`
 
         case "$st" in
-        *perl ) echo '#!/usr/bin/env perl'
+        *perl )  PERL=`which perl`
+                 PERL=`readlink -f ${PERL}`
+                 echo '#!' ${PERL}
                  sed 1d $f
                  ;;
 
-        */sh )   echo '#!' ${POSIX_SHELL}
+        */sh )   SH=`readlink -f ${POSIX_SHELL}`
+                 echo '#!' ${SH}
                  sed 1d $f
                  ;;
 
-- 
2.11.1

